### ISSUE TITLE
ngOnInit() called every time Observable.interval() fires

### ISSUE THREAD

**Issue by Flo12345** [angular/angular#22711]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

I initialize an Observer.interval() inside ngOnInit() which reduces a time every second by one second. The problem here is, that ngOnInit() is called every time the interval fires, meaning every second:

<pre><code>
import { Component, OnInit, Input } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import 'rxjs/add/observable/interval';
import 'rxjs/add/operator/map';
import { LegacyButton } from '../../interfaces';

import { LegacyButtonService } from '../../legacy-button.service';

@Component({
  selector: 'pf-single-button',
  templateUrl: './single-button.component.html',
  styleUrls: ['./single-button.component.css']
})
export class SingleButtonComponent implements OnInit {

  @Input() buttonData: LegacyButton;

  public blockingTimeLeft: number = 0;
  public countDown;
  
  constructor() {
  }

  private handleBlockingTime(): void {
    this.blockingTimeLeft = this.blockingTimeLeft - 1;
  }

  ngOnInit() {
    this.blockingTimeLeft = +this.buttonData.blockingTime;

    this.countDown = Observable
      .interval( 1000 )
      .take( this.blockingTimeLeft )
      .map( () => { this.handleBlockingTime(); } );
  }

}
</code></pre>

In my component.html, I subscribe to the observable via the async pipe:

<pre><code>
{{ countDown | async }}
</code></pre>

## Expected behavior

ngOnInit() should be called only once, initializing the countDown.

## What is the motivation / use case for changing the behavior?

Docs clearly states that ngOnInit() is called only once.

## Environment

<pre><code>
Angular version: 5.2.8

Browser:
- [x] Chrome (desktop) version 64
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 8.9.0  
- Platform: Windows 

Others:

</code></pre>

**Comment by briancodes** [comment]
Hi. Maybe create a StackBlitz reproduction of the issue and add the link

**Comment by ciesielskico** [comment]
How do you know that ngOnInit is called every second? Whats the issue?

**Comment by ciesielskico** [comment]
I cannot see anything if you dont provide a reproduction.

**Comment by mlc-mlapis** [comment]
@Flo12345 ... create a Stackblitz demo reproduction ... it is supposed as a necessity on each reported issue if it technically possible. Use https://stackblitz.com.

**Comment by Flo12345** [comment]
Sorry for the late reply... but I fixed it myself.
The problem occurred because I had the mentioned component nested in a grid-list with a *ngFor on its tiles:

<code><pre>
<mat-grid-list cols="{{ getNumberOfButtons() }}" rowHeight="fit">
    <mat-grid-tile *ngFor="let button of getButtonData()"
                   [colspan]="1"
                   [rowspan]="1">
      <pf-single-button class="singleButton" [buttonData]="button"></pf-single-button>
    </mat-grid-tile>
  </mat-grid-list>
</pre></code>

The method "getButtonData()" itself returned a input-parameter from the parent-component. Turned out, that this is updated on each change-detection. Also, each "pf-single-button" is injected on each change-detection cycle, resulting in the mentioned behaviour.

I solved the problem by changing the method "getButtonData()" to return a component-internal variable, which is set to the input-parameter inside the parents "ngOnInit()" method.
